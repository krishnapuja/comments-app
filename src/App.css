.pad-10 {
  padding: 10px;
}

.margin-30 {
  margin: 30px;
}

.color-grey {
  color: grey;
}

.flex-grow-1 {
  flex-grow: 1;
}

/* Modal css */

/* Modal Window */
.modal {
  position: fixed;
  z-index: 1;
  padding-top: 100px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.4);
}

/* Modal Content */
.modal-content {
  background-color: #fefefe;
  margin: auto;
  padding: 20px;
  border: 1px solid #888;
  width: 80%;
}

/* Modla Close Button */
.close {
  color: #aaaaaa;
  float: right;
  font-size: 28px;
  font-weight: bold;
}

.close:hover,
.close:focus {
  color: #000;
  text-decoration: none;
  cursor: pointer;
}



/* notification  badge css */

.notification {
  position: relative;
  display: inline-block;
}

.notification .badge {
  position: absolute;
  top: -5px;
  right: -5px;
  padding: 5px 10px;
  border-radius: 50%;
  background: white;
}


/*notification-menu css */

.dropdown-content {
  position: absolute;
  right: 0px;
  background-color: #f9f9f9;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
}

.dropdown-content p {
  color: black;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.dropdown-content p:hover {
  background-color: #f1f1f1;
}

.notification:hover .dropdown-content {
  display: block;
}


/*add-comment-page */

.add-comment-heading {
  margin-left: 30px;
  color: #a4c991;
}

.input-field {
  width: 100%;
  padding: 12px;
  margin: 10px 0px;
}

.button-container {
  display: flex;
  justify-content: center;
  align-items: center;
}

.add-comment-button {
  background-color: #a4c991;
  border: 1px solid grey;
  border-radius: 5px;
  margin: 10px;
}

/* Comment card page */
.message-content {
  font-size: large;
}

.comment-card {
  border: 1px solid #d3d3d3;
  margin: 20px;
  padding: 10px;
}

/* Comment Feed page */
.comments-feed-header {
  background-color: #a4c991;
  padding: 10px 20px;
  display: flex;
  align-items: center;
}

@media only screen and (min-width: 768px) {
  .main-section {
    margin: 0 10% 0 10%;
  }
}