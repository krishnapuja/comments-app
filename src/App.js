import logo from './logo.svg';
import './App.css';
import CommentsFeedContainer from './components/CommentsFeedContainer';

function App() {
  return (
 <CommentsFeedContainer />
  );
}

export default App;
