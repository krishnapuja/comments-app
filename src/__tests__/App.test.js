import { render } from '@testing-library/react';
import App from '../App';

describe('render App correctly', () => {
  test('renders the App snapshot correctly', () => {
    const Apprenderer = render(<App />);
    expect(Apprenderer).toMatchSnapshot();
  });
});
