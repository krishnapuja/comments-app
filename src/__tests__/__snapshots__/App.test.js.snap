// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App correctly renders the App snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="comments-feed-header"
      >
        <h2
          class="flex-grow-1"
        >
          Comments Feed
        </h2>
        <div
          class="notification"
        >
          <img
            class="pad-10"
            src="notification-icon.png"
          />
          <div
            class="dropdown-content"
            style="display: none;"
          />
        </div>
        <img
          src="add-comment-icon.png"
          style="padding: 10px;"
        />
      </header>
      <main
        class="main-section"
      >
        <div
          class="modal"
          style="display: none;"
        >
          <div
            class="modal-content"
          >
            <span
              class="close"
            >
              ×
            </span>
            <h3
              class="add-comment-heading "
            >
              Add your Comment
            </h3>
            <form
              class="margin-30"
            >
              <label
                for="name"
              >
                Name
              </label>
              <br />
              <input
                class="input-field"
                id="name"
                name="name"
                type="text"
                value=""
              />
              <br />
              <label
                for="message"
              >
                Message
              </label>
              <br />
              <textarea
                class="input-field"
                cols="50"
                id="message"
                name="message"
                rows="4"
              />
               
              <br />
              <div
                class="button-container"
              >
                <button
                  class="add-comment-button pad-10"
                >
                  Add Comment
                </button>
              </div>
            </form>
          </div>
        </div>
      </main>
      <main
        class="main-section"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="comments-feed-header"
    >
      <h2
        class="flex-grow-1"
      >
        Comments Feed
      </h2>
      <div
        class="notification"
      >
        <img
          class="pad-10"
          src="notification-icon.png"
        />
        <div
          class="dropdown-content"
          style="display: none;"
        />
      </div>
      <img
        src="add-comment-icon.png"
        style="padding: 10px;"
      />
    </header>
    <main
      class="main-section"
    >
      <div
        class="modal"
        style="display: none;"
      >
        <div
          class="modal-content"
        >
          <span
            class="close"
          >
            ×
          </span>
          <h3
            class="add-comment-heading "
          >
            Add your Comment
          </h3>
          <form
            class="margin-30"
          >
            <label
              for="name"
            >
              Name
            </label>
            <br />
            <input
              class="input-field"
              id="name"
              name="name"
              type="text"
              value=""
            />
            <br />
            <label
              for="message"
            >
              Message
            </label>
            <br />
            <textarea
              class="input-field"
              cols="50"
              id="message"
              name="message"
              rows="4"
            />
             
            <br />
            <div
              class="button-container"
            >
              <button
                class="add-comment-button pad-10"
              >
                Add Comment
              </button>
            </div>
          </form>
        </div>
      </div>
    </main>
    <main
      class="main-section"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
